src/main/java/cubicchunks/worldgen/GeneratorPipeline.java

```diff
@@ -25,31 +25,33 @@
 
 import cubicchunks.CubicChunks;
 import cubicchunks.server.ServerCubeCache;
-import cubicchunks.util.AddressTools;
-import cubicchunks.util.Progress;
+import cubicchunks.util.CubeCoords;
 import cubicchunks.util.processor.CubeProcessor;
-import cubicchunks.util.processor.QueueProcessor;
 import cubicchunks.world.cube.Cube;
 import cubicchunks.world.dependency.CubeDependency;
 import cubicchunks.worldgen.dependency.DependentCube;
 import cubicchunks.worldgen.dependency.DependentCubeManager;
 
 import java.util.ArrayList;
 import java.util.HashMap;
+import java.util.LinkedList;
 import java.util.List;
 import java.util.Map;
+import java.util.Queue;
 
 public class GeneratorPipeline {
 
-	private static final int TickBudget = 40; // ms. There are only 50 ms per tick
+	private static final int MAX_CUBES_PER_TICK = 500;
```

Comment:
Regarding the maximum, this was the old value and I just stuck with it. I didn't see lots of "can't keep up" messages during testing, so I assumed its fine. What number do you suggest? I guess ultimately some form of automatic adaptation to the server performance might be best. On my rig (which is likely overpowered to serve as a good benchmark) managed ~2500-3000 processed cubes/s.